source/NVDAObjects/window/excel.py

```diff
@@ -1017,8 +1010,140 @@ def _getFormatFieldAndOffsets(self,offset,formatConfig,calculateOffsets=True):
 		return self.obj.getCellPosition()
 
+NVCELLINFOFLAG_ADDRESS=0x1
+NVCELLINFOFLAG_TEXT=0x2
+NVCELLINFOFLAG_INPUTMESSAGE=0x4
+NVCELLINFOFLAG_STATES=0x8
+NVCELLINFOFLAG_COORDS=0x10
+NVCELLINFOFLAG_OUTLINELEVEL=0x20
+NVCELLINFOFLAG_COMMENTS=0x40
+NVCELLINFOFLAG_FORMULA=0x80
+NVCELLINFOFLAG_ALL=0xffff
+
+class ExcelCellInfo(ctypes.Structure):
+		_fields_=[
+			('text',comtypes.BSTR),
+			('address',comtypes.BSTR),
+			('inputTitle',comtypes.BSTR),
+			('inputMessage',comtypes.BSTR),
+			('states',ctypes.c_longlong),
+			('rowNumber',ctypes.c_long),
+			('rowSpan',ctypes.c_long),
+			('columnNumber',ctypes.c_long),
+			('columnSpan',ctypes.c_long),
+			('outlineLevel',ctypes.c_long),
+			('comments',comtypes.BSTR),
+			('formula',comtypes.BSTR),
+		]
+
+class ExcelCellInfoQuickNavItem(browseMode.QuickNavItem):
+
+	def __init__( self , parentIterator, cellInfo):
+		self.excelCellInfo = cellInfo
+		self.parentIterator=parentIterator
+		super( ExcelCellInfoQuickNavItem ,self).__init__( parentIterator.itemType , parentIterator.document )
+
+	def activate(self):
+		pass
+
+	def isChild(self,parent):
+		return False
+
+	def report(self,readUnit=None):
+		pass
+
+	def __lt__(self,other):
+		return (self.excelCellInfo.rowNumber,self.excelCellInfo.columnNumber)<(other.excelCellInfo.rowNumber,other.excelCellInfo.columnNumber)
+
+	def moveTo(self):
+		cell=self.parentIterator.document.excelWorksheetObject.cells(self.excelCellInfo.rowNumber,self.excelCellInfo.columnNumber)
+		cell.Activate()
+		eventHandler.queueEvent("gainFocus",api.getDesktopObject().objectWithFocus())
+
+	@property
+	def isAfterSelection(self):
+		activeCell=self.parentIterator.selectedCellInfo
+		return (self.excelCellInfo.rowNumber,self.excelCellInfo.columnNumber)>(activeCell.rowNumber,activeCell.columnNumber)
+
+	@property
+	def label(self):
+		return "%s: %s"%(self.excelCellInfo.address.split('!')[-1],self.excelCellInfo.text)
+
+class CommentExcelCellInfoQuickNavItem(ExcelCellInfoQuickNavItem):
+
+	@property
+	def label(self):
+		return "%s: %s"%(self.excelCellInfo.address.split('!')[-1],self.excelCellInfo.comments)
+
+class FormulaExcelCellInfoQuickNavItem(ExcelCellInfoQuickNavItem):
+
+	@property
+	def label(self):
+		return "%s: %s"%(self.excelCellInfo.address.split('!')[-1],self.excelCellInfo.formula)
+
+class ExcelCellInfoQuicknavIterator(object):
+	QuickNavItem=None
+	cellInfoFlags=NVCELLINFOFLAG_ADDRESS+NVCELLINFOFLAG_COORDS
+
+	def __init__(self, itemType , document , direction , includeCurrent):
+		"""
+		See L{QuickNavItemIterator} for itemType, document and direction definitions.
+		@ param includeCurrent: if true then any item at the initial position will be also emitted rather than just further ones. 
+		"""
+		self.document=document
+		self.itemType=itemType
+		self.direction=direction if direction else "next"
+		self.includeCurrent=includeCurrent
+		self.selectedCellInfo=self.document._getSelection().excelCellInfo
+
+	def collectionFromWorksheet(self,worksheetObject):

```

Comment:
Could you add abc.ABCMeta as a metaclass and then decorate this with abc.abstractmethod?

Reply:
Done.